airflow/providers/amazon/aws/sensors/sqs.py

```diff
@@ -69,31 +89,48 @@ def poke(self, context):
 
         self.log.info('SQSSensor checking for message on queue: %s', self.sqs_queue)
 
-        messages = sqs_conn.receive_message(
-            QueueUrl=self.sqs_queue,
-            MaxNumberOfMessages=self.max_messages,
-            WaitTimeSeconds=self.wait_time_seconds,
-        )
+        receive_message_kwargs = {
+            'QueueUrl': self.sqs_queue,
+            'MaxNumberOfMessages': self.max_messages,
+            'WaitTimeSeconds': self.wait_time_seconds,
+        }
+        if self.visibility_timeout is not None:
+            receive_message_kwargs['VisibilityTimeout'] = self.visibility_timeout
+
+        response = sqs_conn.receive_message(**receive_message_kwargs)
+
+        if "Messages" not in response:
+            return False
 
-        self.log.info("received message %s", str(messages))
+        messages = response['Messages']
+        num_messages = len(messages)
+        self.log.info("received %s messages", str(num_messages))
 
-        if 'Messages' in messages and messages['Messages']:
-            entries = [
-                {'Id': message['MessageId'], 'ReceiptHandle': message['ReceiptHandle']}
-                for message in messages['Messages']
-            ]
+        if num_messages == 0:
```

Comment:
```suggestion
        if not num_messages:
```

Same for other checks below.

Reply:
Done, thank you